authentication/pipeline/user.py

```diff
@@ -108,29 +108,27 @@ def create_user_via_email(
             errors=["Full name must be at least 2 characters long."],
         )
 
-    data["email"] = kwargs.get("email", kwargs.get("details", {}).get("email"))
+    data["email"] = (
+        kwargs.get("email", kwargs.get("details", {}).get("email")).strip().lower()
+    )
+
+    if User.objects.filter(email__iexact=data["email"]).exists():
+        raise InvalidEmail(backend)
+
     username = usernameify(data["name"], email=data["email"])
+    data["username"] = username
 
     affiliate_id = get_affiliate_id_from_request(strategy.request)
     if affiliate_id is not None:
         context["affiliate_id"] = affiliate_id
 
-    try:
-        user = User.objects.get(email=data["email"])
-        serializer = UserSerializer(user, data=data, context=context)
-    except User.DoesNotExist:
-        data["username"] = username
-        serializer = UserSerializer(data=data, context=context)
+    serializer = UserSerializer(data=data, context=context)
 
     if not serializer.is_valid():
         raise RequirePasswordAndPersonalInfoException(
             backend, current_partial, errors=serializer.errors
         )
 
-    if user:
-        updated_user = serializer.save()
-        return {"is_new": True, "user": updated_user, "username": updated_user.username}
-
```

Comment:
In the above check where we are checking if it exists, Are we always going to get the user object in this step of the pipeline?

Reply:
I added this check in the previous [PR](https://github.com/mitodl/mitxpro/pull/2588) where we allowed a user to save his information again on Step 1 of the Registration Form. But after this issue is raised, we are not allowing users to go back to step 1 and save the information again. As we talked about before starting work on this issue. So, I've removed this check, because now we don't need this check here.